### PR TITLE
Momchil/various

### PR DESCRIPTION
**Pull request by momchil-flex**

### CODE REVIEW

File: tidy3d/components/base.py [hunk 1]
```diff
@@ -253,7 +253,6 @@ def generate_docstring(cls) -> str:
     # if the model already has a docstring, get the first lines and save the rest
     original_docstrings = []
     if cls.__doc__:
-        original_docstring = cls.__doc__.replace("    ", "\t")
```
Comment (tylerflex) [inline tidy3d/components/base.py:256]: hm.  so does this fix something?

File: tidy3d/components/data.py [hunk 2]
```diff
@@ -848,7 +848,8 @@ def at_centers(self, field_monitor_name: str) -> xr.Dataset:
 
     @equal_aspect
     @add_ax_if_none
-    def plot_field(  # pylint:disable=too-many-arguments, too-many-locals
+    # pylint:disable=too-many-arguments, too-many-locals, too-many-branches, too-many-statements
```
Comment (tylerflex) [inline tidy3d/components/data.py:851]: Yup, summarizes how I feel about this method :D

File: tidy3d/components/data.py [hunk 3]
```diff
@@ -913,8 +914,8 @@ def plot_field(  # pylint:disable=too-many-arguments, too-many-locals
         if field_name == "int":
             monitor_data = self.at_centers(field_monitor_name)
             xr_data = 0.0
-            for field_name in ("Ex", "Ey", "Ez"):
-                field_data = monitor_data[field_name]
+            for field in ("Ex", "Ey", "Ez"):
```
Comment (tylerflex) [inline tidy3d/components/data.py:917]: good call

File: tidy3d/components/simulation.py [hunk 4]
```diff
@@ -6,8 +6,12 @@
 import numpy as np
 import xarray as xr
 import matplotlib.pylab as plt
-import matplotlib as mpl
-from mpl_toolkits.axes_grid1 import make_axes_locatable
+
+try:
+    import matplotlib as mpl
+    from mpl_toolkits.axes_grid1 import make_axes_locatable
+except Exception:  # pylint: disable=broad-except
```
Comment (tylerflex) [inline tidy3d/components/simulation.py:13]: hm, this is probably not great practice.  do you think eventually we can find a solution?

File: tidy3d/components/base.py (hunk 1 above)
Comment (momchil-flex) [reply tidy3d/components/base.py:256]: Was an unused variable caught by pylint

File: tidy3d/components/data.py (hunk 3 above)
Comment (momchil-flex) [reply tidy3d/components/data.py:917]: again, pylint caught that :)

File: tidy3d/components/simulation.py (hunk 4 above)
Comment (momchil-flex) [reply tidy3d/components/simulation.py:13]: Yes, eventually...

**Pull request merged**